Content.Client/CharacterAppearance/Systems/HumanoidAppearanceSystem.cs

```diff
@@ -59,12 +59,18 @@ private void UpdateLooks(EntityUid uid, HumanoidAppearanceComponent component, C
             }
 
             sprite.LayerSetColor(HumanoidVisualLayers.Hair,
-                component.CanColorHair ? component.Appearance.HairColor : Color.White);
+                (!component.HairMatchesSkin
+                    ? (component.CanColorHair ? component.Appearance.HairColor : Color.White)
+                    : component.Appearance.SkinColor).WithAlpha(component.HairAlpha));
             sprite.LayerSetColor(HumanoidVisualLayers.FacialHair,
-                component.CanColorFacialHair ? component.Appearance.FacialHairColor : Color.White);
+                (!component.HairMatchesSkin
+                    ? (component.CanColorFacialHair ? component.Appearance.FacialHairColor : Color.White)
+                    : component.Appearance.SkinColor).WithAlpha(component.HairAlpha));
```

Comment:
Same here, also with a ternary like this you can flip it so you aren't negating the condition